### PR TITLE
Unregister cr

### PR DESCRIPTION
**Pull request by Houshoupei84**

### CODE REVIEW

File: blockchain/blockvalidator.go [hunk 1]
```diff
@@ -115,6 +116,12 @@ func (b *BlockChain) CheckBlockSanity(block *Block) error {
 	return nil
 }
 
+func getDIDByCode(code []byte) (*Uint168, error) {
+	pk := code[1 : len(code)-1]
```
Comment (peterwillcn) [inline blockchain/blockvalidator.go]: code[1 : len(code)-1] => code[0 : len(code)-1] ?

File: blockchain/blockvalidator.go [hunk 2]
```diff
@@ -115,6 +116,12 @@ func (b *BlockChain) CheckBlockSanity(block *Block) error {
...
+	did, error := contract.PublicKeyToDepositProgramHash(pk)
```
Comment (raindust) [inline blockchain/blockvalidator.go]: PublicKeyToDepositProgramHash can't return a did program hash

File: blockchain/blockvalidator_test.go [hunk 3]
```diff
@@ -4,7 +4,10 @@ import (
 	"bytes"
 	"encoding/hex"
 	"fmt"
+	"github.com/elastos/Elastos.ELA/core/types/payload"
```
Comment (raindust) [inline blockchain/blockvalidator_test.go]: format imports

Comment (Houshoupei84) [reply blockchain/blockvalidator_test.go]: fixed

File: blockchain/blockvalidator.go (hunk 2 above)
Comment (Houshoupei84) [reply blockchain/blockvalidator.go]: fixed

**Pull request merged**